### ISSUE TITLE
Pierre allows a PR to self-reference

### ISSUE THREAD

**Issue by mosheeshel** [alvarocavalcanti/pierre-decheck#43]
Should not be able to do that, creates a cyclic reference

**Comment by alvarocavalcanti** [comment]
Good catch @mosheeshel thanks!

**Comment by rajalakshmi-v15** [comment]
Hi.
I have difficulty in understanding what this issue and which section of the code base this issue deals with.
I am new to the open source environment so please give me some directions.
Thanks :)

**Comment by alvarocavalcanti** [comment]
Alright @rajalakshmi-v15 I can help you with that.

First of all, this issue is about cyclic dependency. Given a PR with id XYZ, if one adds "Depends on #XYZ" as a comment, _Pierre_ will include it as a dependency that most likely will make developers wonder what's wrong, since it's depending on itself to be met.

As for where in the code this can be fixed, well, there are some places, depending on the strategy one would think is the best:

1. Have a look at [get_dependency_state](https://github.com/alvarocavalcanti/pierre-decheck/blob/master/lib/pierre.py#L196). That's where _Pierre_ retrieves each dependency's status (it's either `open` or `closed` per GitHub specification) and could be used to consider always the current PR status as `closed`.
1. One could simply ignore the dependency from the start, and that could be accomplished on [get_dependencies_from_bodies](https://github.com/alvarocavalcanti/pierre-decheck/blob/master/lib/pierre.py#L153)

Or if you have another suggestion, we can discuss it here. Be my guest. 😁

**Comment by rajalakshmi-v15** [comment]
Thank you, @alvarocavalcanti (and sorry for the delayed response). I would like to work on this issue.

**Comment by mosheeshel** [comment]
@rajalakshmi-v15 go ahead!

**Comment by rajalakshmi-v15** [comment]
Hi, @mosheeshel. Is there a way to get the id of the current PR? If we can get that, that particular id can be excluded from the `set(dependencies)`  part in `get_dependencies_from_bodies` function.

**Comment by mosheeshel** [comment]
Have a look in [payloads](https://github.com/alvarocavalcanti/pierre-decheck/blob/master/lib/payloads.py) you can see some samples of what's returned by github. I'm guessing you can find some useful identifications.

**Comment by rajalakshmi-v15** [comment]
@mosheeshel, I have opened a new pull request with the branches aligined.
Kindly review it.

**Comment by alvarocavalcanti** [comment]
It was fixed by #52
